test/widgets/app_test.dart

```diff
@@ -361,5 +362,24 @@ void main() {
       await tester.pumpAndSettle();
       check(findSnackBarByText('unrelated').evaluate()).single;
     });
+
+    testWidgets('reportErrorToUserModally', (tester) async {
+      addTearDown(testBinding.reset);
+      await tester.pumpWidget(const ZulipApp());
+      const message = 'test error message';
+      const details = 'details';
+
+      // Prior to app startup, reportErrorToUserModally only logs.
+      reportErrorToUserModally(message, details: details);
+      check(ZulipApp.ready).value.isFalse();
+      await tester.pump();
+      check(find.byType(AlertDialog)).findsNothing();
```

Comment:
I have a commit

```
test [nfc]: Factor out checkNoErrorDialog helper
```

in my revision for #1239; how about cherry-picking that as a prep commit and using it here? (See reasoning in the commit)